### ISSUE TITLE
Add Automatic special-measure Configuration Based on Readout Pulses

### ISSUE THREAD

**Issue by terrorfisch** [qutech/qupulse#63]
The library should generate a MATLAB file which defines a special-measure scan.

The scan configures an Alazar PCI digitizer and the AtsAverage.dll which preprocesses and reduces the acquired data on-the-fly based on so called masks which define the measurement windows. It can perform different operations such as downsampling or repetitive signal averaging.

For this we need to:
1. Identify the measurement windows
2. Compute the mask parameters(the user has to choose a suitable mask type, f.i. a periodic mask)
3. Genereate a file which defines a scan struct.
4. (Run the scan)

Questions:
- Should the library provide the possibility to control MATLAB from there using the MATLAB python interface?
- Where happens the configuration of the operations(downsampling, repetitive signal averaging, etc.)?
- Do we need measurement window 'flagging'? This could be useful to assign different operations to different measurement windows.
- Where happens the data anlysis? 

We can possibly configure and run the whole scan from python if the operation configuration is also done there and the library 'owns' the MATLAB engine.

**Comment by lumip** [comment]
For the identification of measurement windows, this depends on issue #66 .
Further, this issue is related to issue #64 , which is also concerced with Python-MATLAB interaction., as well as issue #65 , concerning the definition of scanlines in the toolkit. These efforts should be coordinated. Especially, I think this issue depends on the results of the ones mentioned above and implementation should thus be postponed.
However, since this is also relevant for the other ticket, answers to the questions should be given. I'll start with some guesses (in order corresponding to the questions above) and hope someone from @qutech/owners @qutech/bluhm-group can elaborate:
- Ultimately, I think it would be useful to run the entire experiment from Python which would then need to control the MATLAB components. However, I assume this comes down to personal preferences and both approaches seem reasonable. We could implement the software in a way that allows for both. This would also allows for easier transition since all parts that are not yet implemented in python could then be run in MATLAB and we hand control over to the Python part when it's done.
- If everything else is configured in Python later on, this should be too. However, until we implement this (which might take some time), manual editing of the created MATLAB files might be a temporary solution. However, I can only guess here, since I don't know exactly, how these operations are used/how often they are changed/whether they depend on the experiment or stay the same for a pulse template etc. There is not so much thought we put into the data acquisition part of the Python software yet and we might need to coordinate with some potential users for detailed requirements.
- same as before
- As far as I know, all current data analysis is done in MATLAB. It seems reasonable to not force everyone to port their algorithms to Python. However, if data evaluation results change parameters, an interface is used. This might be implemented with some specialized Parameter subclasses. Of course, implementing data evaluation entirely in Python is also feasible.

**Comment by terrorfisch** [comment]
I will close this now as the scope of qupulse was narrowed

> Should the library provide the possibility to control MATLAB from there using the MATLAB python interface?

No.
 > Where happens the configuration of the operations(downsampling, repetitive signal averaging, etc.)?

Not a qupulse task.
 > Do we need measurement window 'flagging'? This could be useful to assign different operations to different measurement windows.

Measurement windows have names
 > Where happens the data anlysis?

Not in qupulse.
